Snakefile

```diff
@@ -65,16 +65,24 @@ if config["custom_rules"] is not []:
         include: rule
 
 
-rule clean:
+rule clean_monte:
     run:
-        shell("snakemake -j 1 solve_all_networks --delete-all-output")
         try:
             shell("snakemake -j 1 solve_all_networks_monte --delete-all-output")
         except:
             pass
         shell("snakemake -j 1 run_all_scenarios --delete-all-output")
 
 
+rule clean:
```

Comment:
The original clean also cleaned the monte carlo.
Do you think a new rule is needed?

Reply:
```bash
rule clean:
    run:
        shell("snakemake -j 1 solve_all_networks --delete-all-output")
        try:
            shell("snakemake -j 1 solve_all_networks_monte --delete-all-output")
        except:
            pass
        shell("snakemake -j 1 run_all_scenarios --delete-all-output")
```
This is the original clean. The first `snakemake` doesn't get executed.
Here is the error log in the terminal
```
Restricted license - for non-production use only - expires 2025-11-24
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pulp/apis/gurobi_api.py:170: UserWarning: GUROBI error: Unknown parameter '_test'.
  warnings.warn("GUROBI error: {}.".format(e))
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pypsa/networkclustering.py:16: UserWarning: The namespace `pypsa.networkclustering` is deprecated and will be removed in PyPSA v0.24. Please use `pypsa.clustering.spatial instead`. 
  warnings.warn(
Building DAG of jobs...
WARNING:snakemake.logging:Building DAG of jobs...
Using shell: /bin/bash
WARNING:snakemake.logging:Using shell: /bin/bash
Provided cores: 1 (use --cores to define parallelism)
WARNING:snakemake.logging:Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
WARNING:snakemake.logging:Rules claiming more threads will be scaled down.
Job stats:
job      count
-----  -------
clean        1
total        1

WARNING:snakemake.logging:Job stats:
job      count
-----  -------
clean        1
total        1

Select jobs to execute...
WARNING:snakemake.logging:Select jobs to execute...

INFO:snakemake.logging:
[Fri Jan 12 11:26:39 2024]
INFO:snakemake.logging:[Fri Jan 12 11:26:39 2024]
rule clean:
    jobid: 0
    reason: Rules with neither input nor output files are always executed.
    resources: tmpdir=/var/folders/c3/657p__892k72qxh6r8fdkqsh0000gn/T
INFO:snakemake.logging:rule clean:
    jobid: 0
    reason: Rules with neither input nor output files are always executed.
    resources: tmpdir=/var/folders/c3/657p__892k72qxh6r8fdkqsh0000gn/T

INFO:snakemake.logging:
Restricted license - for non-production use only - expires 2025-11-24
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pulp/apis/gurobi_api.py:170: UserWarning: GUROBI error: Unknown parameter '_test'.
  warnings.warn("GUROBI error: {}.".format(e))
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pypsa/networkclustering.py:16: UserWarning: The namespace `pypsa.networkclustering` is deprecated and will be removed in PyPSA v0.24. Please use `pypsa.clustering.spatial instead`. 
  warnings.warn(
Building DAG of jobs...
WARNING:snakemake.logging:Building DAG of jobs...
Using shell: /bin/bash
WARNING:snakemake.logging:Using shell: /bin/bash
Provided cores: 1 (use --cores to define parallelism)
WARNING:snakemake.logging:Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
WARNING:snakemake.logging:Rules claiming more threads will be scaled down.
Select jobs to execute...
WARNING:snakemake.logging:Select jobs to execute...
Restricted license - for non-production use only - expires 2025-11-24
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pulp/apis/gurobi_api.py:170: UserWarning: GUROBI error: Unknown parameter '_test'.
  warnings.warn("GUROBI error: {}.".format(e))
/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/site-packages/pypsa/networkclustering.py:16: UserWarning: The namespace `pypsa.networkclustering` is deprecated and will be removed in PyPSA v0.24. Please use `pypsa.clustering.spatial instead`. 
  warnings.warn(
Building DAG of jobs...
WARNING:snakemake.logging:Building DAG of jobs...
MissingInputException in rule solve_all_networks in file /Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile, line 102:
Missing input files for rule solve_all_networks:
    affected files:
        results/NG/networks/elec_s_10_ec_lcopt_Co2L-24H.nc
ERROR:snakemake.logging:MissingInputException in rule solve_all_networks in file /Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile, line 102:
Missing input files for rule solve_all_networks:
    affected files:
        results/NG/networks/elec_s_10_ec_lcopt_Co2L-24H.nc
[Fri Jan 12 11:26:43 2024]
INFO:snakemake.logging:[Fri Jan 12 11:26:43 2024]
Error in rule clean:
    jobid: 0

ERROR:snakemake.logging:Error in rule clean:
    jobid: 0

RuleException:
CalledProcessError in file /Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile, line 69:
Command 'set -euo pipefail;  snakemake -j 1 solve_all_networks --delete-all-output' returned non-zero exit status 1.
  File "/Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile", line 69, in __rule_clean
  File "/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/concurrent/futures/thread.py", line 58, in run
ERROR:snakemake.logging:RuleException:
CalledProcessError in file /Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile, line 69:
Command 'set -euo pipefail;  snakemake -j 1 solve_all_networks --delete-all-output' returned non-zero exit status 1.
  File "/Users/gbotemi/Documents/code/PYPSA/mc/pypsa-earth/Snakefile", line 69, in __rule_clean
  File "/Users/gbotemi/miniconda3/envs/pypsa-earth-v2/lib/python3.10/concurrent/futures/thread.py", line 58, in run
Shutting down, this might take some time.
WARNING:snakemake.logging:Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
ERROR:snakemake.logging:Exiting because a job execution failed. Look above for error message
Shutting down, this might take some time.
WARNING:snakemake.logging:Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
ERROR:snakemake.logging:Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-01-12T112636.677565.snakemake.log
WARNING:snakemake.logging:Complete log: .snakemake/log/2024-01-12T112636.677565.snakemake.log
```

Reply:
We might not need a new rule. Modifying the existing rule should be sufficient.

Reply:
Mmmm interesting, may you try removing the -j 1 in the commands? that may solve the error

Reply:
I tried that. It didn't work. The -j 1 command is required at the very least to make the command work.

Reply:
I have modified the clean rule. The clean rule now works for cleaning monte-carlo.